gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -175,7 +180,10 @@ public static GapicContext parse(CodeGeneratorRequest request) {
             mixinServices,
             transport);
 
-    Preconditions.checkState(!services.isEmpty(), "No services found to generate");
+    if (services.isEmpty()) {
+      LOGGER.warning("No services found to generate. This will produce an empty generated srcjar");
+      return GapicContext.empty();
+    }
```

Comment:
qq, if the proto files passed to the generator only have messages (no services), will the srcjar still empty?

Reply:
Yes. In theory you could pass a list of message protos, but if there are no services then it will noop

Reply:
I modified this to not generate a srcjar at all if no services are found